flink-java/src/main/java/org/apache/flink/api/java/typeutils/runtime/AvroSerializer.java

```diff
@@ -91,7 +98,22 @@ public T createInstance() {
 	@Override
 	public T copy(T from) {
 		checkKryoInitialized();
-		return this.kryo.copy(from);
+		try {
+			return this.kryo.copy(from);
+		} catch(KryoException ke) {
+			// kryo was unable to copy it, so we do it through serialization:
+			ByteArrayOutputStream baout = new ByteArrayOutputStream();
+			Output output = new Output(baout);
+
+			kryo.writeObject(output, from);
+
+			output.close();
+
+			ByteArrayInputStream bain = new ByteArrayInputStream(baout.toByteArray());
+			Input input = new Input(bain);
+
+			return (T)kryo.readObject(input, from.getClass());
```

Comment:
Some Kryo serializers do not implement the copy method. In that case, we do a serialization copy.